WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasSnippetButton.lua

```diff
@@ -0,0 +1,228 @@
+  tooltip:SetOwner(frame, "ANCHOR_RIGHT")
+  tooltip:ClearLines()
+  if frame.editable then
+    tooltip:AddDoubleLine(frame.titleText, L["(Right click to rename)"], nil, nil, nil, 0.6, 0.6, 0.6)
+  else
+    tooltip:AddLine(frame.titleText)
+  end
+  tooltip:AddLine("   ")
+  tooltip:AddLine(frame.descriptionText, 0.8, 0.8, 0.8)
+  tooltip:Show()
+  frame.obj:Fire("OnEnter")
+end
+
+local function Control_OnLeave(frame)
+  GameTooltip:Hide()
+  frame.obj:Fire("OnLeave")
+end
+
+local function rename_complete(self, ...)
+  self:ClearFocus()
+  AceGUI:ClearFocus()
+  self:Disable()
+  self:Hide()
+  self:GetParent().obj:Fire("OnEnterPressed", ...)
+end
+--[[-----------------------------------------------------------------------------
+Methods
+-------------------------------------------------------------------------------]]
+local methods = {
+  ["OnAcquire"] = function(self)
+    -- restore default values
+    self:SetDisabled(false)
+    self:SetTitle()
+    self:SetEditable(false)
+  end,
+  -- ["OnRelease"] = nil,
+
+  ["SetTitle"] = function(self, text)
+    self.frame.titleText = text
+    self.title:SetText(text)
+  end,
+  ["SetDescription"] = function(self, text)
+    self.frame.descriptionText = text
+  end,
+  ["SetDisabled"] = function(self, disabled)
+    self.disabled = disabled
+    if disabled then
+      self.frame:Disable()
+    else
+      self.frame:Enable()
+    end
+  end,
+  ["LockHighlight"] = function(self)
+    self.frame:LockHighlight()
+  end,
+  ["UnlockHighlight"] = function(self)
+    self.frame:UnlockHighlight()
+  end,
+  ["SetEditable"] = function(self, bool)
+    if bool then
+      self.frame.editable = true
+      self.deleteButton:Show()
+      self.title:SetPoint("LEFT", self.deleteButton, "RIGHT")
+    else
+      self.frame.editable = false
+      self.deleteButton:Hide()
+      self.title:SetPoint("LEFT", self.deleteButton, "LEFT", 4, 0)
+    end
+  end
+}
+
+--[[-----------------------------------------------------------------------------
+Constructor
+-------------------------------------------------------------------------------]]
+local function Constructor()
+  local name = "WeakAurasSnippetButton" .. AceGUI:GetNextWidgetNum(Type)
+  local button = CreateFrame("Button", name, UIParent, "OptionsListButtonTemplate")
+  button:Hide()
+
+  button:EnableMouse(true)
+  button:SetScript("OnClick", Button_OnClick)
+  button:SetScript("OnEnter", Control_OnEnter)
+  button:SetScript("OnLeave", Control_OnLeave)
+
+  button:SetHeight(24)
+  button:SetWidth(170)
+
+  local deleteButton = CreateFrame("BUTTON", nil, button)
+  deleteButton:SetPoint("LEFT", button, "LEFT")
+  deleteButton:SetSize(20, 20)
+  local deleteTex = deleteButton:CreateTexture()
+  deleteTex:SetAllPoints()
+  deleteTex:SetTexture([[Interface\Buttons\CancelButton-Up]])
+  deleteTex:SetTexCoord(0.1, 0.9, 0.1, 0.9)
+  deleteButton:SetNormalTexture(deleteTex)
+  deleteButton:Hide()
+  button.deleteButton = deleteButton
```

Comment:
The delete button should be on the right side, as that's the more common place for those kind of buttons.